src/gilded_rose.js

```diff
@@ -21,45 +21,56 @@ const items = [
 
 updateQuality(items);
 */
-export function updateQuality(items) {
+export function updateQuality(itemsArray) {
+  let result = []
+  const items = [...itemsArray]
   for (var i = 0; i < items.length; i++) {
-    if (items[i].name === 'Sulfuras, Hand of Ragnaros') {
-      break;
-    }
-    if (items[i].name != 'Aged Brie'
-    && items[i].name != 'Backstage passes to a TAFKAL80ETC concert'
-    && items[i].quality > 0) {
-      items[i].quality = items[i].quality - 1
+    let {name, quality, sell_in} = items[i]
```

Comment:
❤️ Love the deconstruction here! Very nicely done! One thing to keep in mind, though, is that this line is effectively saying that the name, quality, and sell-in values for each of these items are mutable. Meaning that we're allowing mutations on the value of the item. Since `name` doesn't change, this value can actually be a `const`. So why can't this currently be `const {name, quality, sell_in} = items[i]`?

Currently, you're modifying the value of `quality` on line 32 and 37, which is why you're running into runtime errors when assigning a `const` to `name`, `quality`, and `sell_in`. If, however, you update these values to assign a value to a mutable variable instead of overwriting the _original_ value, you can use `const`. For instance:

```
const {name, quality, sell_in} = items[i]
let updatedQuality = quality;
```
In JavaScript, assigning a new variable the value of _another_ variable [does not copy the reference](https://codeburst.io/explaining-value-vs-reference-in-javascript-647a975e12a0) (usually). This means that, for instance:

```
const myFavoriteColor = 'green'
let myCurrentFavoriteColor = myFavoriteColor
myCurrentFavoriteColor = 'blue'

myCurrentFavoriteColor // blue
myFavoriteColor // green
```
This means that you can have `updatedQuality` have the default value of `quality`, but if you update `updatedQuality`, then the value of `quality` isn't mutated.